### DISCUSSION TITLE
Documentation?

### DISCUSSION THREAD

**Discussion by MudkipWithShades** [Techokami/SonicWorldsNext#77] [Ideas]
Is it possible to include documentation of how the project works with the code and properties for nodes? I believe it will be helpful for some people who want to make massive changes to the Sonic engine, but use some mechanics for their own game.

**Comment by DimensionWarped** [comment]
That's a bit of a big ask. Godot doesn't exactly have good tools for generating UML diagrams and the like and if I'm honest a lot of this code wasn't really written with stuff like encapsulation in mind, so you have a lot of access to things that you probably shouldn't from everywhere in the codebase and that makes documentation hard too. I don't think it's the biggest priority right now to get documentation at that depth.

For my own projects from now on though I'm open to using what documentation generation stuff there already is.

https://godotengine.org/article/gsoc-2020-progress-report-1/#gdscript-doc
